app/services/meta-tags.ts

```diff
@@ -0,0 +1,184 @@
+        };
+
+        if (metaTagsOverrides === Object(metaTagsOverrides)) {
+            Object.assign(metaTagsData, metaTagsOverrides);
+        }
+
+        // Include URL, DOI, and any additional identifiers.
+        const identifiers = Array.isArray(metaTagsData.url) ? metaTagsData.url : [metaTagsData.url];
+        if (Array.isArray(metaTagsData.doi)) {
+            identifiers.push(...metaTagsData.doi);
+        } else {
+            identifiers.push(metaTagsData.doi);
+        }
+        if (Array.isArray(metaTagsData.identifier)) {
+            identifiers.push(...metaTagsData.identifier);
+        } else {
+            identifiers.push(metaTagsData.identifier);
+        }
+
+        return {
+            // Citation
+            citation_title: metaTagsData.title,
+            citation_doi: metaTagsData.doi,
+            citation_publisher: metaTagsData.site_name,
+            citation_author_institution: metaTagsData.institution,
+            citation_description: metaTagsData.description,
+            citation_public_url: metaTagsData.url,
+            citation_publication_date: metaTagsData.published_date,
+            // Dublin Core
+            'dc.title': metaTagsData.title,
+            'dc.type': metaTagsData.type,
+            'dc.identifier': identifiers,
+            'dc.abstract': metaTagsData.description,
+            'dc.license': metaTagsData.license,
+            'dc.datemodified': metaTagsData.modified_date,
+            'dc.datesubmitted': metaTagsData.published_date,
+            'dc.publisher': metaTagsData.site_name,
+            'dc.language': metaTagsData.language,
+            // Open Graph/Facebook
+            'og:ttl': 345600, // 4 days = min value.
+            'og:site_name': metaTagsData.site_name,
+            'og:url': metaTagsData.url,
+            'og:description': metaTagsData.description,
+            'og:image': metaTagsData.image,
+            'og:image:type': metaTagsData.image_type,
+            'og:image:width': metaTagsData.image_width,
+            'og:image:height': metaTagsData.image_height,
+            'og:image:alt': metaTagsData.image_alt,
+            // Twitter
+            'twitter:card': 'summary',
+            'twitter:site': metaTagsData.twitter_site,
+            'twitter:creator': metaTagsData.twitter_creator,
+        };
+    }
+
+    /**
+     * Get head tag definitions suitable for the head-tags service.
+     *
+     * @method getHeadTags
+     * @param {MetaTagsData} metaTagsData Data values to use for meta tags.
+     * @return {HeadTagDef[]} Returns head tag defintions.
+     */
+    getHeadTags(this: MetaTags, metaTagsData: MetaTagsData): HeadTagDef[] {
+        const metaTagsDefs: MetaTagsDefs = this.getMetaTags(metaTagsData);
+        const headTagsAttrs: MetaTagAttrs[] = [];
+
+        Object.entries(metaTagsDefs).forEach(([name, content]) => {
+            // Expand lists of meta tag values into multiple tags.
+            if (Array.isArray(content)) {
```

Comment:
If you had the `toArray` thing, this could be simplified:
```ts
Object.entries(metaTagsDefs).flatMap(
    ([name, content]) => toArray(content).map(
        contentMember => this.makeMetaTagAttrs(name, contentMember)
    )
)
```
aw, but `flatMap` doesn't exist yet... I guess it could be a `map` with a hackier flatten:
```ts
[].concat(...arrayOfArrays)
```